src/jobs/actions/rabbitmqaction.ts

```diff
@@ -12,80 +11,99 @@ export class RabbitMQJobAction<T> implements JobAction<T> {
   private connection;
   private binding;
 
-  constructor(data: Record<string, any>) {
+  constructor(data: Record<string, unknown>) {
     Logger.log(
       "Initializing RabbitMQJobAction. Params: " + JSON.stringify(data),
       "RabbitMQJobAction",
     );
 
+    // Validate that all necessary params are present
+    const requiredConnectionParams = [
+      "hostname",
+      "port",
+      "username",
+      "password",
+    ];
+    for (const param of requiredConnectionParams) {
+      if (!data[param]) {
+        throw new NotFoundException(`Missing connection parameter: ${param}`);
+      }
+    }
+
+    const requiredBindingParams = ["exchange", "queue", "key"];
+    for (const param of requiredBindingParams) {
+      if (!data[param]) {
+        throw new NotFoundException(`Missing binding parameter: ${param}`);
+      }
+    }
     this.connection = {
       protocol: "amqp",
-      hostname: data.hostname,
-      port: data.port,
-      username: data.username,
-      password: data.password,
+      hostname: data.hostname as string,
+      port: data.port as number,
+      username: data.username as string,
+      password: data.password as string,
     };
     this.binding = {
-      exchange: data.exchange,
-      queue: data.queue,
-      key: data.key
+      exchange: data.exchange as string,
+      queue: data.queue as string,
+      key: data.key as string,
     };
   }
 
   getActionType(): string {
     return RabbitMQJobAction.actionType;
   }
 
-  async validate(dto: T) {
-    Logger.log(
-      "Validating RabbitMQJobAction: " + JSON.stringify(dto),
-      "RabbitMQJobAction",
-    );
-
-    const connectionDetailsMissing = [undefined, ""].some(el => Object.values(this.connection).includes(el));
-    if (connectionDetailsMissing) {
-      throw new NotFoundException("RabbitMQ configuration is missing connection details.");
-    }
-
-    const bindingDetailsMissing = [undefined, ""].some(el => Object.values(this.binding).includes(el));
-    if (bindingDetailsMissing) {
-      throw new NotFoundException("RabbitMQ binding is missing exchange/queue/key details.");
-    }
-  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async validate(dto: T) {}
```

Comment:
@sbliven I struggle to understand the need for `validate()` in the `action class`. I was always wondering whether it makes sense validating after calling the constructor, and your changes here I think prove my point that it probably doesn't. I would suggest following this example and moving all validation inside the constructor of each action. Then:
- On job `create` if creating an action fails, the job creation should also fail.
- On job `statusUpdate` things get a bit more complicated. Since we are using a global job configuration, instead of job specific, if no changes were made to the configuration then no validation is needed at all and we can proceed directly to performing the job. But, if the configuration has changed, we would need a method that would first update the values of the class' attributes, before performing the action.

Reply:
After discussing with @sofyalaski, I will create a separate PR to address this.